test/json_diff_ex_test.exs

```diff
@@ -156,6 +156,24 @@ defmodule JsonDiffExTest do
     comparediff(s1, s2, res)
   end
 
+  test "check === with same numeric type" do
+    m1 = %{"1" => 4, "2" => 2}
+    m2 = %{"1" => 4, "2" => 2}
+    assert diff(m1, m2) == %{}
+  end
+
+  test "check == but different numeric type" do
+    m1 = %{"1" => 4, "2" => 2}
+    m2 = %{"1" => 4.0, "2" => 2}
+    assert diff(m1, m2) == %{"1" => [4, 4.0]}
+  end
+
+  test "check === but different numeric type" do
+    m1 = %{"1" => 4, "2" => 2}
+    m2 = %{"1" => 4.0, "2" => 2}
+    assert diff(m1, m2) == %{}
```

Comment:
I was a little bit worried that this would pass but you missed adding the new option ;)